fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/derby/DerbyResourceDAO.java

```diff
@@ -437,7 +423,7 @@ public long storeResource(String tablePrefix, List<ExtractedParameterValue> para
             if (parameters != null) {
                 // Derby doesn't support partitioned multi-tenancy, so we disable it on the DAO:
                 try (ParameterVisitorBatchDAO pvd = new ParameterVisitorBatchDAO(conn, null, tablePrefix, false, v_logical_resource_id, 100,
-                    new ParameterNameCacheAdapter(parameterNameDAO), new CodeSystemCacheAdapter(codeSystemDAO))) {
+                    new ParameterNameCacheAdapter(parameterDao), new CodeSystemCacheAdapter(parameterDao))) {
```

Comment:
why does codeSystemCacheAdapter now get the parameterDao?

Reply:
the parameterDao is actually a wrap for both parameterNameDAO and codeSystemDAO, the the cache implemenation is done inside parameterDao.  very interesting implementation.